node/src/components/consensus/era_supervisor/era.rs

```diff
@@ -211,6 +211,14 @@ impl<I> Era<I> {
             .iter()
             .filter_map(|(public_key, weight)| is_active(public_key).then(|| *weight))
             .sum();
+        if total_weight.is_zero() {
+            error!(total_rewards, "total validator weight is 0");
+            return EraReport {
+                rewards: BTreeMap::new(),
+                equivocators: self.accusations(),
+                inactive_validators,
+            };
+        }
```

Comment:
Slight nitpick - we're probably more interested whether `total_weight >> shift` is 0 rather than whether `total_weight` itself is 0. I guess both cases are error conditions in a way, actually.

Reply:
You're right, I'll change this.

(Technically, the former can't happen without the latter: If `total_weight` is not 0 and `shift` is not 0, then that means `total_weight * total_rewards` overflows, which implies that `total_weight` is greater than 2^448. But that's far from obvious and it's better to check the actual divisor.)

Reply:
Done in https://github.com/casper-network/casper-node/pull/1950/commits/1896cb077ffd2f2e05f46960532c7bbac031d5ce.